File: cpp/test/test_3.cpp
#include "../src/code_3.cpp"
#include "cpp_deps/boilerplate.hpp"

using namespace std;

void test(Solution& sol, const json& input, const json& output) {
    string s = input["s"].get<string>();
    int expected = output.get<int>();
    int result = sol.lengthOfLongestSubstring(s);
    CHECK_EQ(result, expected);
}

TEST_CASE("") {
    TEST(3);
}

File: python/tests/test_3.py
from python.src.code_3 import Solution
from python.tests.python_deps.boilerplate import BoilerTest


def input_fetcher(case):
    return case["input"]["s"]


def test():
    test = BoilerTest(3, input_fetcher)
    solution = Solution().lengthOfLongestSubstring
    test.run_tests(solution)
